From 2b78cf8e91cb684c1ac7fe9fb7cdce87a1f3a7bd Mon Sep 17 00:00:00 2001
From: WeiguangTWK <weiguangtwk@outlook.com>
Date: Wed, 1 Jul 2026 15:08:08 +0800
Subject: [PATCH] Neko LineageOS: Add Push Broker permission & debug package

About LineagePushProviderReference:

It introduces a placeholder package in debug build enabling developer to check if broker in their build works

Change-Id: I3cfe7bc4817b7a763f23056150082f3dc199aa99
---
 config/common.mk                          |  1 +
 config/permissions/org.lineageos.push.xml | 23 +++++++++++++++++++++++
 2 files changed, 24 insertions(+)
 create mode 100644 config/permissions/org.lineageos.push.xml

diff --git a/config/common.mk b/config/common.mk
index 8adef89..96db7a6 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -227,6 +227,7 @@ PRODUCT_PRODUCT_PROPERTIES += \
 
 # These packages are excluded from user builds
 PRODUCT_PACKAGES_DEBUG += \
+    LineagePushProviderReference \
     procmem
 
 ifneq ($(TARGET_BUILD_VARIANT),user)
diff --git a/config/permissions/org.lineageos.push.xml b/config/permissions/org.lineageos.push.xml
new file mode 100644
index 0000000..3b66ea0
--- /dev/null
+++ b/config/permissions/org.lineageos.push.xml
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (C) 2026 Neko-LineageOS
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+
+<permissions>
+    <!-- The feature listed here coincides with the PushBrokerService declared as
+        part of the external lineage service init in SystemServer. If this xml isn't present
+        on the device, the service will not be published as a binder service.
+    -->
+    <feature name="org.lineageos.push" />
+</permissions>
-- 
2.43.0

